synfig-core/src/modules/lyr_freetype/lyr_freetype.cpp

```diff
@@ -314,23 +314,45 @@ struct FaceMetaData
 		return *static_cast<FaceMetaData*>(face->generic.data);
 	}
 
-	void
-	add_to_face(FT_Face face)
+	static void
+	add_to_face(FT_Face face, filesystem::Path path)
 	{
 		if (face->generic.data)
 			face->generic.finalizer(face);
-		face->generic.data = this;
+		face->generic.data = new FaceMetaData{path};
 		face->generic.finalizer = FaceMetaData::self_destroy;
 	}
 
+#if HAVE_HARFBUZZ
+	static void
+	add_to_face(FT_Face face, filesystem::Path path, hb_font_t* font)
```

Comment:
It is better to pass it as `const filesystem::Path&` or implement a move constructor and use move semantics.

Reply:
Done.